pkg/apis/duck/v1alpha1/pubsub_defaults.go

```diff
@@ -17,13 +17,25 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"context"
+
+	"github.com/google/knative-gcp/pkg/apis/configs/gcpauth"
+
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
+	"knative.dev/pkg/apis"
 )
 
-func (s *PubSubSpec) SetPubSubDefaults() {
-	if s.GoogleServiceAccount == "" && s.ServiceAccountName == "" &&
-		(s.Secret == nil || equality.Semantic.DeepEqual(s.Secret, &corev1.SecretKeySelector{})) {
-		s.Secret = DefaultGoogleCloudSecretSelector()
+func (s *PubSubSpec) SetPubSubDefaults(ctx context.Context) {
+	ad := gcpauth.FromContextOrDefaults(ctx).AuthorizationDefaults
+	if ad == nil {
+		// TODO This should probably error out, rather than silently allow in non-defaulted COs.
```

Comment:
can this happen? If the configMap is not there maybe? Shouldn't have failed before?
If not, then maybe we should error out as you said... Or default to the default hardcoded secret values?

Reply:
I haven't seen it happen (outside unit tests), but there is enough magic happening in the pkg code that it might be possible. I added an error log, so at least we might be able to notice something went wrong.

My guess is that we will eventually just make this fail the request. But I want some experience with it before I do that. Is that OK with you?

Reply:
OK with me!